src/core/net/srp_server.cpp

```diff
@@ -1324,6 +1324,50 @@ otError Server::Service::SetFullName(const char *aFullName)
     return error;
 }
 
+otError Server::Service::GetNextTxtEntry(otDnsTxtIterator &aIterator, otDnsTxtEntry &aTxtEntry) const
+{
+    otError error = OT_ERROR_NONE;
+
+    for (uint16_t i = aIterator; i < mTxtLength;)
+    {
+        otDnsTxtEntry entry;
+        uint8_t       length = mTxtData[i++];
+
```

Comment:
Should we do a check hereto make sure that the entire txt entry is within `mTxtData` and it` mTxtLength`?
Maybe something like this:
``` 
  VerifyOrExit(i + length <= mTxtLength, error = OT_ERROR_PARSE);
```
This is to ensure that when we return the `mKey` and `mValue` pointers (and their length) the buffers are all within the `mTxtData` buffer and that we won't read beyond end of buffer.

Reply:
There is another method `VerifyTxtData` that makes sure the TXT data is well-formed, and this method accepts only valid TXT data. The idea is to reject invalid TXT data as a DNS format error before each entry is queried. This makes sure we don't accept a portion of the TXT data. I updated the documentation to make this clear.

Reply:
Nice! Did not notice `VerifyTxtData`. Sounds good to me.